test/integration/protocol_integration_test.cc

```diff
@@ -4922,4 +4922,64 @@ TEST_P(DownstreamProtocolIntegrationTest, UnknownPseudoHeader) {
   }
 }
 
+// This test performs DNS refreshing very fast also keep changing DNS results.
+// It is trying to reproduce a R/W race for logical_dns_cluster.
+class LogicalDnsReadWriteRaceTest : public ProtocolIntegrationTest {
```

Comment:
[This PR](https://github.com/envoyproxy/envoy/pull/32967) moved this test into a separate file, which can make the test run much faster: